src/main/scala/cromwell/engine/CallActor.scala

```diff
@@ -1,81 +1,125 @@
 package cromwell.engine
 
-import akka.actor.{Actor, Props}
-import akka.event.{Logging, LoggingReceive}
+import akka.actor.FSM.NullFunction
+import akka.actor.{LoggingFSM, Props}
 import cromwell.binding._
 import cromwell.binding.values.WdlValue
-import cromwell.engine
-import cromwell.engine.backend.Backend
+import cromwell.engine.CallActor.CallActorState
+import cromwell.engine.backend.{TaskAbortedException, Backend}
 import cromwell.engine.workflow.WorkflowActor
 import cromwell.engine.workflow.WorkflowActor.CallFailed
 
 import scala.language.postfixOps
-import scala.util.{Failure, Success}
-
+import scala.util.{Try, Failure, Success}
 
 object CallActor {
+
   sealed trait CallActorMessage
-  case class Start(inputs: CallInputs) extends CallActorMessage
-  case class ExecutionFinished(outputs: CallOutputs) extends CallActorMessage
-  case class ExecutionFailed(regret: Throwable) extends CallActorMessage
+  case object Start extends CallActorMessage
+  final case class RegisterCallAbortFunction(abortFunction: AbortFunction) extends CallActorMessage
+  case object AbortCall extends CallActorMessage
+  final case class ExecutionFinished(call: Call, outputs: Try[CallOutputs]) extends CallActorMessage
```

Comment:
need, no. it's a nice touch though, see this blog post for an explanation: http://noelwelsh.com/programming/2015/06/04/more-on-sealed/